app/src/main/java/me/guillaumin/android/osmtracker/db/model/TrackStatistics.java

```diff
@@ -0,0 +1,195 @@
+package me.guillaumin.android.osmtracker.db.model;
+
+import me.guillaumin.android.osmtracker.db.TrackContentProvider;
+import me.guillaumin.android.osmtracker.db.TrackContentProvider.Schema;
+import android.content.ContentResolver;
+import android.database.Cursor;
+import android.os.Bundle; 
+
+/**
+ * Represents statistics, such as  total length and maximum speed, for a Track
+ *
+ * @author Arseniy Lartsev
+ *
+ */
+public class TrackStatistics {
+	private long trackId;
+	private long pointCount;
+	private float length;
+	private float maxSpeed;
+	private double lastLatitude;
+	private double lastLongitude;
+	private long lastTime;
+	private int lastId;
+	private float timeMoving;
+	private ContentResolver contentResolver;
+	
+	private static double square(double x) {
+		return x*x;
+	}
+
+	/**
+	 * Compute the approximate distance in meters between two locations, with 0.5% precision
+	 */
+	public static float getDistance(double lat1, double long1, double lat2, double long2) {
+		final double R = 6370e3; // Earth radius
+		final double rad_per_deg = Math.PI/180; // radians per degree
+		
+		lat1 *= rad_per_deg;
+		long1 *= rad_per_deg;
+		lat2 *= rad_per_deg;
+		long2 *= rad_per_deg;
+		
+		double dLat = lat2-lat1;
+		double dLon = long2-long1;
+
+		double A = square(Math.sin(0.5*dLat)) + Math.cos(lat1) * Math.cos(lat2) * square(Math.sin(0.5*dLon));
+		
+		return (float)(2 * R * Math.asin(Math.sqrt(A)));
+	}
+
+	public TrackStatistics () {
+		trackId = -1;
```

Comment:
Actually this constructor seems not use, and can be unsafe as the class requires a `ContentResolver` to work properly, so we'd get a `NullPointerException` later?

I suggest removing it, and making the required fields like `contentResolver` `final` so that there's no chance they can be left uninitialised.